intelmq/bots/parsers/shadowserver/config.py

```diff
@@ -928,3 +936,303 @@ def validate_ip(value):
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+    ],
+    'constant_fields': {
+        'protocol.transport': 'tcp',
+        'classification.type': 'compromised',
+        'feed.code': 'shadowserver-sinkhole-http-referer',
+        #'classification.identifier': 'compromised-website',
+    },
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/Vulnerable-ISAKMP
+vulnerable_isakmp = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+        ('source.port', 'port'),
+    ],
+    'optional_fields': [
+        ('source.reverse_dns', 'hostname'),
+        # ('classification.identifier', 'tag'),  # This will be 'openike' in constant fields
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('source.geolocation.region', 'region'),
+        ('source.geolocation.city', 'city'),
+        ('extra.', 'naics', invalidate_zero),
+        ('extra.', 'sic', invalidate_zero),
+        ('extra.', 'initiator_spi', validate_to_none),
+        ('extra.', 'responder_spi', validate_to_none),
+        ('extra.', 'next_payload', validate_to_none),
+        ('extra.', 'version', validate_to_none),
+        ('extra.', 'exchange_type', validate_to_none),
+        ('extra.', 'flags', validate_to_none),
+        ('extra.', 'message_id', validate_to_none),
+        ('extra.', 'next_payload2', validate_to_none),
+        ('extra.', 'domain_of_interpretation', validate_to_none),
+        ('extra.', 'protocol_id', validate_to_none),
+        ('extra.', 'spi_size', validate_to_none),
+        ('extra.', 'notify_message_type', validate_to_none),
+    ],
+    'constant_fields': {
+        'protocol.transport': 'udp',
+        'classification.type': 'vulnerable service',
+        'feed.code': 'shadowserver-vulnerable-isakmp',
+        'classification.identifier': 'openike',
+    }
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/Botnet-CCIP
+botnet_ccip = {
+    'required_fields': [
+        ('time.source', 'first_seen', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+        ('source.port', 'port'),
+    ],
+    'optional_fields': [
+        ('extra.', 'channel', validate_to_none),
+        ('source.asn', 'asn'),
+        ('source.as_name', 'as_name'),
+        ('source.geolocation.cc', 'geo'),
+        ('source.geolocation.region', 'region'),
+        ('source.geolocation.city', 'city'),
+        ('source.fqdn', 'domain'),
+    ],
+    'constant_fields': {
+        'classification.type': 'c&c',
+        'feed.code': 'shadowserver-botnet-ccip',
+    }
+}
+
+accessible_rdp = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+        ('source.port', 'port'),
+    ],
+    'optional_fields': [
+        ('source.reverse_dns', 'hostname'),
+        # ('classification.identifier', 'tag'),  # This will be 'openrdp' in constant fields
```

Comment:
The latest report we have at this moment (25.10.2016) contains these fields:
```
"timestamp","ip","port","hostname","tag","handshake","asn","geo","region","city","rdp_protocol","cert_length","subject_common_name","issuer_common_name","cert_issue_date","cert_expiration_date","sha1_fingerprint","cert_serial_number","ssl_version","signature_algorithm","key_algorithm","sha256_fingerprint","sha512_fingerprint","md5_fingerprint","naics","sic","sector"
```

`tag` will always be **rdp**, so we put this in constant fields section:
```
'classification.identifier': 'openrdp'
```

Protocols are there too:
```
'protocol.transport': 'tcp',
'protocol.application': 'rdp'
```